hooks/brc20/useBrc20TransferFees.ts

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from 'react';
+
+import { UTXO } from 'types';
+import { brc20TransferEstimateFees } from '../../transactions/brc20';
+
+type CommitValueBreakdown = {
+  commitChainFee: number;
+  revealChainFee: number;
+  revealServiceFee: number;
+  transferChainFee: number;
+  transferUtxoValue: number;
+};
+
+export enum ErrorCode {
+  UTXOS_MISSING = 'UTXOS_MISSING',
+  INSUFFICIENT_FUNDS = 'INSUFFICIENT_FUNDS',
+  INVALID_TICK = 'INVALID_TICK',
+  INVALID_AMOUNT = 'INVALID_AMOUNT',
+  INVALID_FEE_RATE = 'INVALID_FEE_RATE',
+  SERVER_ERROR = 'SERVER_ERROR',
+}
+
+type Props = {
+  addressUtxos: UTXO[] | undefined;
+  tick: string;
+  amount: number;
+  feeRate: number;
+  revealAddress: string;
+};
+
+const validateProps = (props: Props) => {
+  const { addressUtxos, tick, amount, feeRate } = props;
+
+  if (!addressUtxos) {
+    return ErrorCode.UTXOS_MISSING;
+  }
+
+  if (!addressUtxos.length) {
+    return ErrorCode.INSUFFICIENT_FUNDS;
+  }
+
+  if (tick.length !== 4) {
+    return ErrorCode.INVALID_TICK;
+  }
+
+  if (amount <= 0) {
+    return ErrorCode.INVALID_AMOUNT;
+  }
+
+  if (feeRate <= 0) {
+    return ErrorCode.INVALID_FEE_RATE;
+  }
+
+  return null;
+};
+
+/**
+ * Estimates the fees for a BRC-20 1-step transfer
+ * @param addressUtxos - The UTXOs in the bitcoin address which will be used for payment
+ * @param tick - The 4 letter BRC-20 token name
+ * @param amount - The amount of the BRC-20 token to transfer
+ * @param feeRate - The desired fee rate for the transactions
+ * @param revealAddress - The address where the balance of the BRC-20 token lives. This is usually the ordinals address.
+ */
+const useBrc20TransferFees = (props: Props) => {
+  const { addressUtxos, tick, amount, feeRate, revealAddress } = props;
```

Comment:
@victorkirov why do we supply addressUtxos instead of finding it within the core function? like:
https://github.com/secretkeylabs/xverse-core/blob/2356a364559511b04ba331d390c738be1a419dce/transactions/btc.ts#L903

Reply:
and make feeRate optional, with default being the default feeRate?
like: https://github.com/secretkeylabs/xverse-core/blob/2356a364559511b04ba331d390c738be1a419dce/transactions/btc.ts#L784

Reply:
We could do both. The counter points for them which we need to consider are:
- For utxos, we usually have the useFees and useExecute hooks in one component. If they both get UTXOs individually, we would be calling the UTXO endpoint twice for the same data. This allows you to get UTXOs from outside the function and the collection can be reused in other parts of the calling component if needed.
- There's a similar argument for the feerate. Also, the component getting the default fee rate itself is a bit magical and side effecty and it's unclear what will happen if you don't specify a feerate.

I'm happy to build both in though if it would be better 👍

Reply:
I wouldn't consider either as a blocker for this feature, but just want to align on the direction we're moving in.

because at the same time, we're refactoring this kind of logic to go into xverse-core: https://github.com/secretkeylabs/xverse-core/pull/201, to avoid mistakes and double fixing on mobile and extension

Reply:
We should probably move it into core but move the useFeeRate and useAddressUTXOs hooks into core as well and then we can build cacheing into the hooks themselves to avoid multiple retrievals 🤔

Reply:
i think this will be tackled in https://linear.app/xverseapp/project/consolidate-bitcoin-transaction-logic-da21d978a728/ENG ?